actionview/lib/action_view/path_set.rb

```diff
@@ -56,7 +56,7 @@ def exists?(path, prefixes, partial, details, details_key, locals)
 
     private
       def search_combinations(prefixes)
-        prefixes = Array(prefixes)
+        prefixes = Array(prefixes).flat_map { |prefix| [prefix, prefix.split("/").tap(&:shift).join("/")] }.uniq
```

Comment:
The goal of this block is to transform an Array like `["fun/active_storage/attachments/_attachment"]` into an `Array` with duplicates removed consisting of 

* `"fun/active_storage/attachments/_attachment"`
* `"active_storage/attachments/_attachment"`
* `"attachments/_attachment"`

I know there is a more clear and concise way to achieve this outcome, so please share any suggestions you might have. Something like [String#delete_prefix](https://ruby-doc.org/3.3.5/String.html#method-i-delete_prefix), but `prefix.delete_until("/")`

Reply:
Seems reasonable. Sorry I'm not in my codebase atm, but is this just doing one `shift` on each prefix? If so, a prefix of `fun/active_storage/attachments/` would now gain `active_storage/attachments/` but not `attachments/`?

Hoping to circle back to my effected app in a couple weeks and play with your `search_combinations` change.

Reply:
That's a good point, the current implementation only omits a single prefix at a time. I think it might need to recurse until it's out of directory names.